misk-zookeeper/src/main/kotlin/misk/clustering/zookeeper/ZkLeaseManager.kt

```diff
@@ -65,19 +66,17 @@ internal class ZkLeaseManager @Inject internal constructor(
     if (!state.compareAndSet(State.NOT_STARTED, State.RUNNING)) return
     connected.set(false)
 
-    // When a cluster change occurs, recheck each lease to see if it should still be owned
-    cluster.watch {
-      actionQueue.put {
-        handleClusterChange()
-      }
-    }
-
     // When we are disconnected from zk, inform all of the leases that their state is now unknown
     client.value.connectionStateListenable.addListener(ConnectionStateListener { _, state ->
       actionQueue.put {
         handleConnectionStateChanged(state.isConnected)
       }
     })
+
+    taskQueue.schedule(leasePollInterval) {
+      checkAllLeases()
```

Comment:
Can this throw? Since you aren't scheduling with backoff it'll just re-poll at the same interval, but will log an error with an exception which might page which I'm not sure that we want?

Reply:
This cannot throw. There's blanket exception handling here https://github.com/square/misk/blob/master/misk-zookeeper/src/main/kotlin/misk/clustering/zookeeper/ZkLease.kt#L80